### PR TITLE
Thank you screens

### PR DESCRIPTION
**Pull request by jellyfish-tom**
## What does this PR do?

This PR introduces "Airdrop thank you screens":
  1. initial thank you screen, informing user about airdrop
  2. second one - requirements screen, that gives more details about airdrop

#### *Where should the reviewer start?*
Top down I would assume :)
 
#### *Any background context you want to provide?*
Logic to display the flow should still be introduced along with "floating button". I imagine it will be in the top level dashboard (wallets) as responsibility of decision process "if app should display thank you screen and if not what to do" should be taken at the top level and not inside "thank you screens" components.

#### *Any scout cleaning/refactoring?*
not really

#### *Any blog post related to the solution you have used?*
nope

## Todos:
1. when terms and conditions are implemented, they should be plugged to T&C in RequirementsScreen.tsx, line 30
2. when Airdrop Dashboard is implemented it should be plugged to RequirementScreen.tsx, line 26

***Developed on iOS***

- [ ] Checked on iOS
- [ ] Checked on Android

### CODE REVIEW

File: loc/en.js [hunk 1]
```diff
@@ -23,6 +23,33 @@ module.exports = {
...
+    itIsAlive: {
+      subtitle: 'Airdrop is live! Check it out!',
+      description: 'We are happy to announce our first Bitcoin Vault Airdrop for Gold Wallet.',
+      iDontCare: "I don't care",
+      learnMore: 'Learn more',
+    },
+    requirements: {
+      soundsGreat: 'Sounds great!',
+      subtitle: 'Thank you for creating the Bitcoin Vault community.',
+      description:
```
Comment (KedziaPawel) [inline loc/en.js:37]: I would add the amount 250.000 as a variable to change in an easy way for all translations if needed

File: loc/en.js [hunk 2]
```diff
@@ -23,6 +23,33 @@ module.exports = {
...
+        'We are glad that you are a part of it! We want to share $250.000 in BTCV as a reward for being a supportive community.',
+      listHeader: 'Requirements for the Airdrop:',
+      points: {
+        0: '3-Key Vault Wallet',
+        1: 'Minimum amount of BTCV stored in 3-Key Vault Wallet - 5 BTCV*',
```
Comment (KedziaPawel) [inline loc/en.js:42]: same with 5 BTCV

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import { ScreenTemplate, Button, Header } from 'app/components';
+import { Route } from 'app/consts';
+import { ApplicationState } from 'app/state';
+import { actions } from 'app/state/airdrop';
+import { typography, palette } from 'app/styles';
+
+const i18n = require('../../../loc');
+
+interface ActionProps {
+  completeThankYouFlow: Function;
```
Comment (KedziaPawel) [inline src/screens/Airdrop/RequirementsScreen.tsx:16]: I think we have a warning in eslint for using a `Function`, please use full typing

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+type Props = {
+  navigation: StackNavigationProp<any, Route.AirdropRequirements>;
```
Comment (KedziaPawel) [inline src/screens/Airdrop/RequirementsScreen.tsx:20]: shouldn't be `any`

File: src/screens/Airdrop/ThankYouScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+class AirdropThankYouScreen extends Component<Props, {}> {
+  componentDidMount() {
+    const { thankYouSeen, markThankYouSeen } = this.props;
+
+    if (!thankYouSeen) {
+      markThankYouSeen();
+    }
+  }
+
+  navigateBack = () => this.props.navigation.navigate(Route.Dashboard);
```
Comment (KedziaPawel) [inline src/screens/Airdrop/ThankYouScreen.tsx:36]: imho `navigateBack = () => this.props.navigation.pop()` would be more generic

File: src/screens/Airdrop/ThankYouScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      >
+        <Text style={styles.subtitle}>{i18n.airdrop.itIsAlive.subtitle}</Text>
+        <Text style={styles.description}>{i18n.airdrop.itIsAlive.description}</Text>
+      </ScreenTemplate>
+    );
+  }
+}
+
+const mapStateToProps = (state: ApplicationState): MapStateProps => ({
+  thankYouSeen: state.airdrop.thankYouSeen,
```
Comment (KedziaPawel) [inline src/screens/Airdrop/ThankYouScreen.tsx:72]: use selectors

Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx:72]: yeah, I even had them implemented, but resigned. What is there reason to use sleectors here, in such a simple scenario? Am not converting store data, am not calculating anything, not changing form, just reaching out for simple value. Why use selectors here?

Comment (KedziaPawel) [reply src/screens/Airdrop/ThankYouScreen.tsx:72]: https://github.com/reduxjs/reselect#reselect
`Selectors are efficient. A selector is not recomputed unless one of its arguments changes.`
Selectors are caching results so there are more performant and would be easier to reuse them in another selector if needed

Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx:72]: 👍  thanks

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+} & ActionProps;
+
+class AirdropRequirementsScreen extends Component<Props, {}> {
+  onSoundsGreatPress = () => {
+    this.props.completeThankYouFlow();
+    // this.props.navigation.navigate(Route.Dashboard); WIP
```
Comment (marek-siemieniuk-morawski) [inline src/screens/Airdrop/RequirementsScreen.tsx:26]: We don't have any convention agreed upon yet, but personally, I like more TODO comments in cases like this.

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+      >
+        <Text style={styles.subtitle}>{i18n.airdrop.requirements.subtitle}</Text>
+        <Text style={styles.description}>{i18n.airdrop.requirements.description}</Text>
+        <View>
+          <Text style={styles.listHeader}>{i18n.airdrop.requirements.listHeader}</Text>
+          {this.getRequirementsListItems()}
+        </View>
+        <View style={styles.dateInfoContainer}>
+          <Text style={styles.description}>{i18n.airdrop.requirements.dateHeader}</Text>
+          <Text style={styles.date}>19/12/2020 1 P.M GMT +1</Text>
```
Comment (marek-siemieniuk-morawski) [inline src/screens/Airdrop/RequirementsScreen.tsx:70]: I'm guessing it shouldn't be hardcoded as well

File: src/screens/Airdrop/ThankYouScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+import { typography, palette } from 'app/styles';
+
+const i18n = require('../../../loc');
+
+interface MapStateProps {
+  thankYouSeen: boolean;
+}
+
+interface ActionProps {
+  markThankYouSeen: Function;
```
Comment (marek-siemieniuk-morawski) [inline src/screens/Airdrop/ThankYouScreen.tsx:19]: same as Pawel mentioned before - simply to not forget 🙂

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 8 above)
Comment (KedziaPawel) [reply src/screens/Airdrop/RequirementsScreen.tsx:70]: it will be added in my PR

File: loc/en.js (hunk 1 above)
Comment (jellyfish-tom) [reply loc/en.js:37]: done

File: loc/en.js (hunk 2 above)
Comment (jellyfish-tom) [reply loc/en.js]: done

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 3 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx]: done

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 4 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx]: done

File: src/screens/Airdrop/ThankYouScreen.tsx (hunk 5 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx]: done

File: src/screens/Airdrop/ThankYouScreen.tsx (hunk 6 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx]: done

File: src/screens/Airdrop/ThankYouScreen.tsx (hunk 9 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx]: sure, thanks :)

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 7 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx:28]: sure, will change to TODO in that case

Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx]: done

File: src/screens/Airdrop/ThankYouScreen.tsx (hunk 9 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/ThankYouScreen.tsx]: done

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+              <TouchableOpacity onPress={this.onTermsConditionsPress}>
+                <Text style={styles.termsText}>{i18n.airdrop.requirements.termsAndConditions.termsAndConditions}</Text>
+              </TouchableOpacity>
+            </View>
+          </>
+        }
+      >
+        <Text style={styles.subtitle}>{i18n.airdrop.requirements.subtitle}</Text>
+        <Text style={styles.description}>
+          {i18n.formatString(i18n.airdrop.requirements.description, CONST.airdropTotalDollarsToShare)}
```
Comment (KedziaPawel) [inline src/screens/Airdrop/RequirementsScreen.tsx:66]: `{i18n.formatString(i18n.airdrop.requirements.description, { airdropTotalDollarsToShare: CONST.airdropTotalDollarsToShare)}}` would be nice to add meaningful name here so the translators would know what they are transleting. Later in the translations:
`description:
This conversation was marked as resolved by jellyfish-tom
        'We are glad that you are a part of it! We want to share ${airdropTotalDollarsToShare} in BTCV as a reward for being a supportive community.',`

File: src/screens/Airdrop/RequirementsScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+        <View>
+          <Text style={styles.listHeader}>{i18n.airdrop.requirements.listHeader}</Text>
+          <View style={styles.listItem}>
+            <Image source={icons.roundTick} style={styles.image} />
+            <Text style={styles.listText}>{i18n.airdrop.requirements.points[0]}</Text>
+          </View>
+          <View style={styles.listItem}>
+            <Image source={icons.roundTick} style={styles.image} />
+            <Text style={styles.listText}>
+              {i18n.formatString(i18n.airdrop.requirements.points[1], CONST.airdropMinimumBTCVRequired)}
```
Comment (KedziaPawel) [inline src/screens/Airdrop/RequirementsScreen.tsx:77]: same

**Review by KedziaPawel — APPROVED** [review]

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 10 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx]: done

File: src/screens/Airdrop/RequirementsScreen.tsx (hunk 11 above)
Comment (jellyfish-tom) [reply src/screens/Airdrop/RequirementsScreen.tsx]: done

**Pull request merged**